### ISSUE TITLE
Build fails on OpenBSD 6.1 (MacPPC) - undefined reference to `sigwait' in filter.c

### ISSUE THREAD

**Issue by larryhynes** [chneukirchen/mblaze#36]
At least, _I think_ that's the point where it's failing; see full build log below.

(I wanted to test `mshow -x` on a different OS......)

```
cc -g -O2 -Wall -Wno-switch -Wextra -fstack-protector-strong -D_FORTIFY_SOURCE=2 -I/usr/local/include   -c -o rfc2047.o rfc2047.c
cc   maddr.o blaze822.o mymemmem.o mytimegm.o seq.o slurp.o rfc2047.o  -L/usr/local/lib -liconv -o maddr
blaze822.o: In function `blaze822_addr':
/home/larry/mblaze/blaze822.c:219: warning: warning: strcpy() is almost always misused, please use strlcpy()
cc -g -O2 -Wall -Wno-switch -Wextra -fstack-protector-strong -D_FORTIFY_SOURCE=2 -I/usr/local/include   -c -o magrep.o magrep.c
cc -g -O2 -Wall -Wno-switch -Wextra -fstack-protector-strong -D_FORTIFY_SOURCE=2 -I/usr/local/include   -c -o rfc2045.o rfc2045.c
cc   magrep.o blaze822.o mymemmem.o mytimegm.o seq.o slurp.o rfc2047.o rfc2045.o  -L/usr/local/lib -liconv -o magrep
blaze822.o: In function `blaze822_addr':
/home/larry/mblaze/blaze822.c:219: warning: warning: strcpy() is almost always misused, please use strlcpy()
cc -g -O2 -Wall -Wno-switch -Wextra -fstack-protector-strong -D_FORTIFY_SOURCE=2 -I/usr/local/include   -c -o mdate.o mdate.c
cc   mdate.o  -L/usr/local/lib -liconv -o mdate
cc -g -O2 -Wall -Wno-switch -Wextra -fstack-protector-strong -D_FORTIFY_SOURCE=2 -I/usr/local/include   -c -o mdeliver.o mdeliver.c
mdeliver.c: In function 'deliver':
mdeliver.c:82: warning: format '%ld' expects type 'long int', but argument 4 has type 'time_t'
cc   mdeliver.o blaze822.o mymemmem.o mytimegm.o  -L/usr/local/lib -liconv -o mdeliver
blaze822.o: In function `blaze822_addr':
/home/larry/mblaze/blaze822.c:219: warning: warning: strcpy() is almost always misused, please use strlcpy()
cc -g -O2 -Wall -Wno-switch -Wextra -fstack-protector-strong -D_FORTIFY_SOURCE=2 -I/usr/local/include   -c -o mdirs.o mdirs.c
cc   mdirs.o  -L/usr/local/lib -liconv -o mdirs
cc -g -O2 -Wall -Wno-switch -Wextra -fstack-protector-strong -D_FORTIFY_SOURCE=2 -I/usr/local/include   -c -o mexport.o mexport.c
cc   mexport.o blaze822.o mymemmem.o mytimegm.o seq.o slurp.o  -L/usr/local/lib -liconv -o mexport
blaze822.o: In function `blaze822_addr':
/home/larry/mblaze/blaze822.c:219: warning: warning: strcpy() is almost always misused, please use strlcpy()
cc -g -O2 -Wall -Wno-switch -Wextra -fstack-protector-strong -D_FORTIFY_SOURCE=2 -I/usr/local/include   -c -o mflag.o mflag.c
cc   mflag.o blaze822.o mymemmem.o mytimegm.o seq.o slurp.o  -L/usr/local/lib -liconv -o mflag
blaze822.o: In function `blaze822_addr':
/home/larry/mblaze/blaze822.c:219: warning: warning: strcpy() is almost always misused, please use strlcpy()
cc -g -O2 -Wall -Wno-switch -Wextra -fstack-protector-strong -D_FORTIFY_SOURCE=2 -I/usr/local/include   -c -o mgenmid.o mgenmid.c
cc   mgenmid.o blaze822.o mymemmem.o mytimegm.o seq.o slurp.o  -L/usr/local/lib -liconv -o mgenmid
blaze822.o: In function `blaze822_addr':
/home/larry/mblaze/blaze822.c:219: warning: warning: strcpy() is almost always misused, please use strlcpy()
cc -g -O2 -Wall -Wno-switch -Wextra -fstack-protector-strong -D_FORTIFY_SOURCE=2 -I/usr/local/include   -c -o mhdr.o mhdr.c
mhdr.c: In function 'print_date':
mhdr.c:89: warning: format '%ld' expects type 'long int', but argument 2 has type 'time_t'
cc   mhdr.o blaze822.o mymemmem.o mytimegm.o seq.o slurp.o rfc2047.o  -L/usr/local/lib -liconv -o mhdr
blaze822.o: In function `blaze822_addr':
/home/larry/mblaze/blaze822.c:219: warning: warning: strcpy() is almost always misused, please use strlcpy()
cc -g -O2 -Wall -Wno-switch -Wextra -fstack-protector-strong -D_FORTIFY_SOURCE=2 -I/usr/local/include   -c -o minc.o minc.c
cc   minc.o  -L/usr/local/lib -liconv -o minc
cc -g -O2 -Wall -Wno-switch -Wextra -fstack-protector-strong -D_FORTIFY_SOURCE=2 -I/usr/local/include   -c -o mlist.o mlist.c
cc   mlist.o seq.o slurp.o  -L/usr/local/lib -liconv -o mlist
cc -g -O2 -Wall -Wno-switch -Wextra -fstack-protector-strong -D_FORTIFY_SOURCE=2 -I/usr/local/include   -c -o mmime.o mmime.c
cc   mmime.o slurp.o  -L/usr/local/lib -liconv -o mmime
cc -g -O2 -Wall -Wno-switch -Wextra -fstack-protector-strong -D_FORTIFY_SOURCE=2 -I/usr/local/include   -c -o mpick.o mpick.c
mpick.c: In function 'parse_dur':
mpick.c:529: warning: missing initializer
mpick.c:529: warning: (near initialization for 'tm.tm_min')
mpick.c: In function 'parse_cmp':
mpick.c:331: warning: 'op' may be used uninitialized in this function
mpick.c:331: note: 'op' was declared here
mpick.c: In function 'main':
mpick.c:688: warning: 'flag' may be used uninitialized in this function
mpick.c:688: note: 'flag' was declared here
cc   mpick.o blaze822.o mymemmem.o mytimegm.o seq.o slurp.o rfc2047.o  -L/usr/local/lib -liconv -o mpick
blaze822.o: In function `blaze822_addr':
/home/larry/mblaze/blaze822.c:219: warning: warning: strcpy() is almost always misused, please use strlcpy()
cc -g -O2 -Wall -Wno-switch -Wextra -fstack-protector-strong -D_FORTIFY_SOURCE=2 -I/usr/local/include   -c -o mscan.o mscan.c
cc -g -O2 -Wall -Wno-switch -Wextra -fstack-protector-strong -D_FORTIFY_SOURCE=2 -I/usr/local/include   -c -o pipeto.o pipeto.c
cc   mscan.o blaze822.o mymemmem.o mytimegm.o seq.o slurp.o rfc2047.o pipeto.o  -L/usr/local/lib -liconv -o mscan
blaze822.o: In function `blaze822_addr':
/home/larry/mblaze/blaze822.c:219: warning: warning: strcpy() is almost always misused, please use strlcpy()
cc -g -O2 -Wall -Wno-switch -Wextra -fstack-protector-strong -D_FORTIFY_SOURCE=2 -I/usr/local/include   -c -o msed.o msed.c
cc   msed.o blaze822.o mymemmem.o mytimegm.o seq.o slurp.o  -L/usr/local/lib -liconv -o msed
blaze822.o: In function `blaze822_addr':
/home/larry/mblaze/blaze822.c:219: warning: warning: strcpy() is almost always misused, please use strlcpy()
cc -g -O2 -Wall -Wno-switch -Wextra -fstack-protector-strong -D_FORTIFY_SOURCE=2 -I/usr/local/include   -c -o mseq.o mseq.c
mseq.c: In function 'main':
mseq.c:311: warning: missing initializer
mseq.c:311: warning: (near initialization for 'iter.cur')
cc   mseq.o seq.o slurp.o  -L/usr/local/lib -liconv -o mseq
cc -g -O2 -Wall -Wno-switch -Wextra -fstack-protector-strong -D_FORTIFY_SOURCE=2 -I/usr/local/include   -c -o mshow.o mshow.c
cc -g -O2 -Wall -Wno-switch -Wextra -fstack-protector-strong -D_FORTIFY_SOURCE=2 -I/usr/local/include   -c -o filter.o filter.c
cc -g -O2 -Wall -Wno-switch -Wextra -fstack-protector-strong -D_FORTIFY_SOURCE=2 -I/usr/local/include   -c -o safe_u8putstr.o safe_u8putstr.c
cc -g -O2 -Wall -Wno-switch -Wextra -fstack-protector-strong -D_FORTIFY_SOURCE=2 -I/usr/local/include   -c -o rfc2231.o rfc2231.c
cc   mshow.o blaze822.o mymemmem.o mytimegm.o seq.o slurp.o rfc2047.o rfc2045.o filter.o safe_u8putstr.o rfc2231.o pipeto.o  -L/usr/local/lib -liconv -o mshow
blaze822.o: In function `blaze822_addr':
/home/larry/mblaze/blaze822.c:219: warning: warning: strcpy() is almost always misused, please use strlcpy()
filter.o: In function `filter':
/home/larry/mblaze/filter.c:120: undefined reference to `sigwait'
collect2: ld returned 1 exit status
gmake: *** [<builtin>: mshow] Error 1
```

**Comment by chneukirchen** [comment]
Try adding `-pthread` to CFLAGS.

Will commit if this works.

**Comment by larryhynes** [comment]
No change with `-pthread` added to CFLAGS, I'm afraid.

```
(....lots of output, leading to....)
cc -pthread -g -O2 -Wall -Wno-switch -Wextra -fstack-protector-strong -D_FORTIFY_SOURCE=2 -I/usr/local/include   -c -o rfc2231.o rfc2231.c
cc   mshow.o blaze822.o mymemmem.o mytimegm.o seq.o slurp.o rfc2047.o rfc2045.o filter.o safe_u8putstr.o rfc2231.o pipeto.o  -L/usr/local/lib -liconv -o mshow
blaze822.o: In function `blaze822_addr':
/home/larry/mblaze/blaze822.c:219: warning: warning: strcpy() is almost always misused, please use strlcpy()
filter.o: In function `filter':
/home/larry/mblaze/filter.c:120: undefined reference to `sigwait'
collect2: ld returned 1 exit status
gmake: *** [<builtin>: mshow] Error 1
```

**Comment by chneukirchen** [comment]
To LDFLAGS too.

Fixed in 1ad29c5

**Comment by larryhynes** [comment]
Thank you. 😄
